plugins/azure/virtualmachines/vmManagedDisks.js

```diff
@@ -0,0 +1,48 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure/');
+
+module.exports = {
+    title: 'VM Managed Disks Enabled',
+    category: 'Virtual Machines',
+    description: 'Ensures that Azure virtual machines are configured to use Azure managed disks.',
+    more_info: 'Azure managed disks are block-level storage volumes that are managed by Azure are like physical disks in an on-premises server but, virtualized. Azure managed disks provide high durability and security.',
+    recommended_action: 'Migrate virtual machine disks to Azure managed disks',
+    link: '',
+    apis: ['virtualMachines:listAll'],
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var locations = helpers.locations(settings.govcloud);
+
+        async.each(locations.virtualMachines, function(location, rcb) {
+            var virtualMachines = helpers.addSource(cache, source,
+                ['virtualMachines', 'listAll', location]);
+
+            if (!virtualMachines) return rcb();
+
+            if (virtualMachines.err || !virtualMachines.data) {
+                helpers.addResult(results, 3, 'Unable to query for virtualMachines: ' + helpers.addError(virtualMachines), location);
+                return rcb();
+            }
+
+            if (!virtualMachines.data.length) {
+                helpers.addResult(results, 0, 'No existing Virtual Machines found', location);
+                return rcb();
+            }
+
+            for (let virtualMachine of virtualMachines.data) {
+                if (virtualMachine.storageProfile && virtualMachine.storageProfile.osDisk &&
```

Comment:
We should also check whether data disk type is also managed

Reply:
@AkhtarAmir We cannot attach an unmanaged data disk if VM is using managed disks and vice versa. So checking OS disk only will ensure that what kind of disks VM is using.